src/script/engine.zig

```diff
@@ -358,6 +363,82 @@ pub const Engine = struct {
         try self.stack.pushByteArray(value);
     }
 
+    /// OP_NIP: Removes the second-to-top stack item
+    ///
+    /// # Returns
+    /// /// - "EngineError.StackUnderflow": if initial stack length < 2
+    fn opNip(self: *Engine) !void {
+        if (self.stack.len() < 2) {
+            return error.StackUnderflow;
+        }
+        const top_value = try self.stack.pop();
+        const second_to_top_value = try self.stack.pop();
+        try self.stack.pushByteArray(top_value);
+        defer self.allocator.free(top_value);
+        defer self.allocator.free(second_to_top_value);
+    }
+
+    /// OP_OVER: Copies the second-to-top stack item to the top
+    ///
+    /// /// # Returns
+    /// - "EngineError.StackUnderflow": if initial stack length < 2
+    fn opOver(self: *Engine) !void {
+        if (self.stack.len() < 2) {
+            return error.StackUnderflow;
+        }
+        const value = try self.stack.peek(1);
+        try self.stack.pushByteArray(value);
+    }
+
+    /// OP_SWAP: The top two items on the stack are swapped.
+    ///
+    /// /// # Returns
+    /// - "EngineError.StackUnderflow": if initial stack length < 2
+    fn opSwap(self: *Engine) !void {
+        if (self.stack.len() < 2) {
+            return error.StackUnderflow;
+        }
+        const top_value = try self.stack.pop();
+        const second_to_top_value = try self.stack.pop();
+        try self.stack.pushByteArray(top_value);
+        try self.stack.pushByteArray(second_to_top_value);
+        defer self.allocator.free(top_value);
+        defer self.allocator.free(second_to_top_value);
+    }
+
+    /// OP_TUCK: The item at the top of the stack is copied and inserted before the second-to-top item.
+    ///
+    /// /// # Returns
+    /// - "EngineError.StackUnderflow": if initial stack length < 2
+    fn opTuck(self: *Engine) !void {
+        if (self.stack.len() < 2) {
+            return error.StackUnderflow;
+        }
+        const top_value = try self.stack.pop();
+        const second_to_top_value = try self.stack.pop();
+        try self.stack.pushByteArray(second_to_top_value);
+        try self.stack.pushByteArray(top_value);
+        try self.stack.pushByteArray(second_to_top_value);
+        defer self.allocator.free(top_value);
+        defer self.allocator.free(second_to_top_value);
```

Comment:
same, don't free when not necessary.
Here you just need 1 alloc and 0 free

Reply:
https://github.com/zig-bitcoin/btczee/pull/34#discussion_r1761130553